rascal-lsp/src/main/rascal/lsp/lang/rascal/lsp/IDECheckerWrapper.rsc

```diff
@@ -53,7 +53,29 @@ import lang::rascalcore::check::ModuleLocations;
 }
 map[loc, set[Message]] checkFile(loc l, set[loc] workspaceFolders, start[Module](loc file) getParseTree, PathConfig(loc file) getPathConfig)
     = job("Rascal check", map[loc, set[Message]](void(str, int) step) {
-    checkForImports = [getParseTree(l)];
+
+    tuple[start[Module], set[Message]] getParseTreeOrErrors(loc l, str name, loc errorLocation) {
+        try {
+            t = getParseTree(l);
+            errors = hasParseErrors(t)
+                ? {error("Cannot typecheck this module, since dependency `<name>` has parse error(s).", errorLocation,
+                        causes=[error("Parse error around this position.", e.src) | e <- findBestParseErrors(t)])}
+                : {};
```

Comment:
it we throw a `ParseError` here, we can reuse the same code of the `catch` a few lines lower and reduce the duplication.